src/modern/class/ReactComponentBase.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright 2013-2014, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactComponentBase
+ */
+
+"use strict";
+
+var ReactInstanceMap = require('ReactInstanceMap');
+
+var assign = require('Object.assign');
+var invariant = require('invariant');
+var warning = require('warning');
+
+/**
+ * Base class helpers for the updating state of a component.
+ */
+function ReactComponentBase(props) {
```

Comment:
I don't fully trust the transpilers that we use to be efficient since classes have to deal with other edge cases. For example, we might start including a runtime in the transpiler.

They may also not always be spec compliant. It might not be equivalent since class methods are probably going to become non-enumerable which we can't do in IE8. But that has gone back and forth.